pkgs/top-level/ocaml-packages.nix

```
@@ -591,6 +591,8 @@ let
 
     result = callPackage ../development/ocaml-modules/ocaml-result { };
 
+    secp256k1 = callPackage ../development/ocaml-modules/secp256k1 { };
```

This package should not receive itself as argument but the C library; here is a possible fix.

```suggestion
    secp256k1 = callPackage ../development/ocaml-modules/secp256k1 {
      inherit (pkgs) secp256k1;
    };
```

---

Good catch @vbgl, thank you!